File: 14_longestCommonPrefix.py
# 14. Longest Common Prefix

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs)==1:
            return strs[0]
        
        for i in range(len(strs)):
            if not strs[i]:
                return ""
            
        prefix = ""
        ind = 0
        l = []
        
        min_l = 1000
        
        for i in range(len(strs)):
            if len(strs[i])<min_l:
                min_l=len(strs[i])
         
        point = 0
        prefix = ""
        while point<min_l:
            for i in range(1, len(strs)):
                if strs[0][point]!=strs[i][point]:
                    return prefix
            prefix+=strs[0][point]
            point+=1
        return prefix
            

File: 14_longestCommonPrefix.cpp
// 14. Longest Common Prefix

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        if(strs.size()==1)
            return strs[0];
        
        int min_l = INT_MAX-1;
 
        for(int i=0;i<strs.size();i++)
            min_l = min(min_l, (int)strs[i].length());
            
        int p = 0;
        int i;
        for(i=0;i<min_l;i++){
            char c = strs[0][i];
            for(int j=1;j<strs.size();j++){
                if(strs[j][i]!=c)
                {
                    int l = (int)(strs[j]).length();
                    return strs[j].substr(0,i);
                }
                    
            }
        }
        
        return strs[0].substr(0,i);
        
    }
};
